python/ray/util/spark/cluster_init.py

```diff
@@ -476,11 +506,44 @@ def _setup_ray_cluster(
 
     cluster_unique_id = uuid.uuid4().hex[:8]
 
+    if (
+        is_in_databricks_runtime()
+        and global_mode_enabled()
+        and ray_temp_root_dir is not None
+    ):
+        _logger.warning(
+            "The `ray_temp_root_dir` argument is ignored when enabling global mode"
+            "on Databricks. We use /local_disk0/tmp as default ray temp root dir."
+        )
+        ray_temp_root_dir = None
     if ray_temp_root_dir is None:
-        ray_temp_root_dir = start_hook.get_default_temp_dir()
-    ray_temp_dir = os.path.join(
-        ray_temp_root_dir, f"ray-{ray_head_port}-{cluster_unique_id}"
-    )
+        ray_temp_root_dir = start_hook.get_default_temp_root_dir()
+    if global_mode_enabled():
+        ray_addr = None
+        try:
+            # It reads from environment variable RAY_ADDRESS, then check
+            # /local_disk0/tmp/ray/ray_current_cluster to see if the address
+            # is available, if yes then a global ray cluster is running, otherwise
+            # it will start a new glbal ray cluster.
+            ray_addr = canonicalize_bootstrap_address_or_die(
+                "auto", get_user_temp_dir()
+            )
+        except Exception:
+            # Surpress exceptions thrown when there's no valid ray address
+            pass
+        if ray_addr is not None:
+            raise RuntimeError(
+                "Global mode is enabled by setting "
+                f"{DATABRICKS_RAY_CLUSTER_GLOBAL_MODE} to true. Current active ray "
+                "cluster on spark haven't shut down. Please call "
+                "`ray.util.spark.shutdown_ray_cluster()` before initiating a "
+                "new Ray cluster on spark."
+            )
+        ray_temp_dir = os.path.join(ray_temp_root_dir, "ray")
```

Comment:
As discussed ,for global mode, we should clean exiting (if it exists) `ray_temp_dir` before starting ray global cluster

Reply:
I cleaned that in start_ray_node.py https://github.com/ray-project/ray/blob/51e74e6c337bd625e3361acd445d8d829c0a6aa0/python/ray/util/spark/start_ray_node.py#L50-L52